apps/prairielearn/src/sync/fromDisk/assessments.ts

```diff
@@ -267,34 +274,38 @@ function getParamsForAssessment(
     type: assessment.type,
     number: assessment.number,
     title: assessment.title,
-    multiple_instance: assessment.multipleInstance ? true : false,
+    multiple_instance: assessment.multipleInstance,
+    // If shuffleQuestions is not set, it's implicitly false for Homework and true for Exams.
     shuffle_questions:
-      (assessment.type === 'Exam' && assessment.shuffleQuestions === undefined) ||
+      (assessment.type === 'Exam' && assessment.shuffleQuestions == null) ||
       assessment.shuffleQuestions
         ? true
         : false,
     allow_issue_reporting: allowIssueReporting,
     allow_real_time_grading: allowRealTimeGrading,
     allow_personal_notes: allowPersonalNotes,
-    require_honor_code: requireHonorCode,
+    // If requireHonorCode is not set, it's implicitly false for Homework and true for Exams.
+    // NOTE: There are various homeworks with requireHonorCode set to true, but that value is ignored.
```

Comment:
Interesting, I was gonna say I didn't think this was possible:

https://github.com/PrairieLearn/PrairieLearn/blob/cd460964fdf3f939b839d4d5d437dcb2720c7a39/apps/prairielearn/src/sync/course-db.ts#L1032-L1034

But that's presumably because we only just recently added that validation and older courses haven't synced recently.

Reply:
Moreso that the previous code did `?? true` for `requireHonorCode` when sending it to the database. https://github.com/PrairieLearn/PrairieLearn/blob/cd460964fdf3f939b839d4d5d437dcb2720c7a39/apps/prairielearn/src/sync/fromDisk/assessments.ts#L57